crypto/keys/ed25519/ed25519_test.go

```diff
@@ -1,6 +1,7 @@
-package ed25519_test
+package ed25519
```

Comment:
why?

Reply:
to avoid weird package renames. It simplifies reading the code.
Also, we only have unit tests here. Using `ed25519_test` is good for integration tests.

Reply:
This is a weird package rename. It actually helps the reader to have _test packages marked consistently.

> is good for integration tests.

Why? How? In which cases?

Please revert this unnecesary change.

Reply:
BTW there are plenty of good reasons to hide internals when testing Go packages, that's why `_test` packages are idiomatic in Go. We've made efforts in the past to make tests live within their respective `_test` packages. This is a step backwards.

Reply:
Testing in same package is the Go idiomatic way. The [go test](https://golang.org/pkg/cmd/go/internal/test/) roughly mentions about `_test` package. The [testing](https://golang.org/pkg/testing/) godoc doesn't mention it at all. Go code review [guidelines](https://github.com/golang/go/wiki/CodeReviewComments#import-dot) mention `_test` packages only in one place - to use it to avoid circular dependencies - they suggest to use `dot import` for that.

In all projects I was working (many) there was a preference to test in the package (not `_test`). Personally I'm advocating to use use `_test` packages for integration tests to have better feeling of a _black box_ testing. This also gives a better separation between unit tests and other tests.

Reply:
@robert-zaremba do you actually need to access any private variables here? If not, I would prefer to stick with `_test` as well.

Reply:
I checked of few popular repositories:

+ [docker ce](https://github.com/docker/docker-ce.git): 3 test files use `_test` pacakge, vs 822 test files which don't.
+ [hugo](https://github.com/gohugoio/hugo.git): 11 test files use `_test` pacakge, vs 271 test files which don't.
+ [cockroach db](https://github.com/cockroachdb/cockroach.git) 386 test files use `_test` pacakge, vs 784 test files which don't.
+ [BigCache](https://github.com/allegro/bigcache.git) 1 test files use `_test` pacakge, vs 11 test files which don't.

Reply:
I don't think statistics are really the important thing here. It was `_test` before and we don't need private variables so why introduce an unneeded line to the PR diff?

Reply:
Because I need to import the `ed25519` package from stdlib for the cross-package signature verification test which was missing. 

In fact I firstly and aliased the new import to `stded25519` and then changed the package name. In secp tests we had: ` 	underlyingSecp256k1 "github.com/btcsuite/btcd/btcec" ` which is also a hack for the name conflict.

Anyway - I explained the main motivation in the first paragraph of the comment [above](https://github.com/cosmos/cosmos-sdk/pull/7672#discussion_r512093963). 

@alessio , @aaronc  - If you prefer to keep the `_test` name, how about using the dot import (ref: same comment above)
 to  avoid this name clashes ?

Reply:
How about just using import aliases? It's pretty much what we've been doing across the whole codebase all along.

Reply:
It's not about which projects use and don't use `_test`. We should strive to use when possible for better isolation.

Reply:
I wanted to point out that it's a usual Go way, especially for unit tests.
So, how about this: 
> @alessio , @aaronc - If you prefer to keep the _test name, how about using the dot import (ref: same comment above) to avoid this name clashes ?

Reply:
> I wanted to point out that it's a usual Go way for unit tests.

And I would like to point out that:

> Test files that declare a package with the suffix "_test" will be compiled as a separate package, and then linked and run with the main test binary.

Per [this piece of documentation here](https://tip.golang.org/cmd/go/#hdr-Test_packages), that incidentally is hosted at `tip.golang.org`.

>  how about using the dot import

No, it adds no value, plus it would make the imported symbols unpredictable.

How about reverting this change?

Reply:
> Per this piece of documentation here, that incidentally is hosted at tip.golang.org.

I already linked `go test` docs (using the source [file](https://golang.org/pkg/cmd/go/internal/test/)) in one of my comments above. I don't see any relevant change in _tip_. If there is any, would you mind to pointing it out?

Reply:
> How about reverting this change?

OK, my motivation was to use the more adopted way for Go unit tests. If this is not an SDK way, no problem, reverting.